### ISSUE TITLE
Esky not building package with dependencies on windows.

### ISSUE THREAD

**Issue by F483** [cloudmatrix/esky#86]
I'm trying to get the simplest example from the tutorial to work.
https://github.com/cloudmatrix/esky/tree/master/tutorial/stage1

My environment:

```
Z:\share_space\esky-master\tutorial\stage1>python --version
Python 3.4.3

Z:\share_space\esky-master\tutorial\stage1>pip freeze
...
cx-Freeze==4.3.4
esky==0.9.8
py2exe==0.9.2.2
...
```

When run using py2exe as the freezer, it fails to find the py2exe module for some reason (though freezing other apps with "python setup.py py2exe" works just fine).

```
Z:\share_space\esky-master\tutorial\stage1>python setup.py bdist_esky
running bdist_esky
Traceback (most recent call last):
  File "setup.py", line 17, in <module>
    "freezer_module":"py2exe",
  File "C:\Python34\lib\distutils\core.py", line 148, in setup
    dist.run_commands()
  File "C:\Python34\lib\distutils\dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "C:\Python34\lib\distutils\dist.py", line 973, in run_command
    cmd_obj.ensure_finalized()
  File "C:\Python34\lib\distutils\cmd.py", line 107, in ensure_finalized
    self.finalize_options()
  File "C:\Python34\lib\site-packages\esky\bdist_esky\__init__.py", line 291, in
 finalize_options
    raise RuntimeError(err)
RuntimeError: freezer module not found: 'py2exe'
```

With cx-Freeze building the package works well but running the executable fails.

```
Z:\share_space\esky-master\tutorial\stage1>dist\example-app-0.1.win32\example.ex
e
ValueError: bad marshal data (unknown type code)
Fatal Python error: unable to locate initialization module

Current thread 0x00001254 (most recent call first):
```

**Comment by timeyyy** [comment]
are you using esky from pypi? It's not up to date. Try cloning the master branch.

**Comment by F483** [comment]
I just tried it on a fresh windows vm with the latest master branch and got the same result.

	E:\esky\tutorial\stage1>python setup.py bdist_esky
	running bdist_esky
	Traceback (most recent call last):
	  File "setup.py", line 17, in <module>
	    "freezer_module":"py2exe",
	  File "C:\Python34\lib\distutils\core.py", line 148, in setup
	    dist.run_commands()
	  File "C:\Python34\lib\distutils\dist.py", line 955, in run_commands
	    self.run_command(cmd)
	  File "C:\Python34\lib\distutils\dist.py", line 973, in run_command
	    cmd_obj.ensure_finalized()
	  File "C:\Python34\lib\distutils\cmd.py", line 107, in ensure_finalized
	    self.finalize_options()
	  File "C:\Python34\lib\site-packages\esky-0.9.9.dev0-py3.4.egg\esky\bdist_esky\
	__init__.py", line 294, in finalize_options
	    raise RuntimeError(err)
	RuntimeError: freezer module not found: 'py2exe'

	E:\esky\tutorial\stage1>pip freeze
	...
	esky==0.9.9.dev0
	py2exe==0.9.2.2
	...

	E:\esky\tutorial\stage1>python --version
	Python 3.4.3

**Comment by F483** [comment]
If I use the freezer cx-Freeze it works however. So it must have something to do with the py2exe module. Having it work with cx-Freeze is good enough for me however.

**Comment by timeyyy** [comment]
could you rename the issue to account for that?

**Comment by F483** [comment]
Renamed accordingly.

**Comment by F483** [comment]
Even using cx-Freeze it seem dependencies are not included.

At this point I'm quite stuck, is there any combination over versions for python, esky, py2exe|cx-Freeze that will let me build a windows distribution with dependencies included. My package works on python 2.7, 3.3, 3.4?

**Comment by timeyyy** [comment]
can we get this in a new issue? which dependencies exactly? can you paste your setup file.

**Comment by F483** [comment]
Well I think the issue is still valid as I cant get any useful build with esky for windows. The project I'm trying to build is this one https://github.com/storj/dataserv-client

The project works on python 2.7, 3.3 and 3.4 as you can see with travis and I have previously build releases with py2exe (see https://github.com/Storj/dataserv-client/releases/tag/v1.1.0 ), so nothing wrong with py2exe.

I would be happy to know if anyone managed to build anything on windows with dependencies and which versions of python, esky and py2exe or cx-Freeze were used.

**Comment by timeyyy** [comment]
sorry but the issue is 100% unrelated to py2exe or pyhton3.4, its regarding eskies lacking documentation and your setup file... i don't wanna clutter up the posts for people that wanna work on the issue, this ran for me http://pastebin.com/UX9m0vNJ

**Comment by F483** [comment]
Ok I changed the title to not include py2exe or python3.4

The setup file you pasted does not work for me. Can you paste your output for `python --version` and `pip freeze`. Also maybe post the generated output.

    E:\dataserv-client>python setup.py bdist_esky
    running bdist_esky
    Traceback (most recent call last):
      File "setup.py", line 103, in <module>
        "Topic :: Software Development :: Libraries :: Python Modules",
      File "C:\Python34\lib\distutils\core.py", line 148, in setup
        dist.run_commands()
      File "C:\Python34\lib\distutils\dist.py", line 955, in run_commands
        self.run_command(cmd)
      File "C:\Python34\lib\distutils\dist.py", line 973, in run_command
        cmd_obj.ensure_finalized()
      File "C:\Python34\lib\distutils\cmd.py", line 107, in ensure_finalized
        self.finalize_options()
      File "C:\Python34\lib\site-packages\esky-0.9.9.dev0-py3.4.egg\esky\bdist_esky\
    __init__.py", line 282, in finalize_options
        raise RuntimeError(err)
    RuntimeError: no supported freezer modules found (try installing bbfreeze)

    E:\dataserv-client>pip freeze
    You are using pip version 6.0.8, however version 7.1.0 is available.
    You should consider upgrading via the 'pip install --upgrade pip' command.
    cannot determine version of editable source in e:\dataserv-client (git command n
    ot found in path)
    -e dataserv-client==1.2.1
    esky==0.9.9.dev0
    future==0.15.0
    py2exe==0.9.2.2
    pycrypto==2.6.1
    randomio==0.2.1

    E:\dataserv-client>python --version
    Python 3.4.3

**Comment by F483** [comment]
I created a branch branch where you can find the [setup.py](https://github.com/Storj/dataserv-client/blob/esky_autoupdate/setup.py) file used.

**Comment by timeyyy** [comment]
python 3.4 and cxfreeze

**Comment by F483** [comment]
When I use cx-Freeze it builds, but does not include the dependencies, as I have already mentioned, see pasted output below.

If it is working for you I would like to know the exact setup you have, so I can replicate it. Could you please post your output for `python --version`, `pip --freeze` and what version of windows you are using.

    E:\dataserv-client\dist\dataserv-client-1.2.1.win32>dataserv-client.exe
    Traceback (most recent call last):
      File "<string>", line 747, in <module>
      File "<string>", line 336, in bootstrap
      File "<string>", line 363, in chainload
      File "<string>", line 733, in _chainload
      File "C:\Python34\lib\site-packages\cx_Freeze\initscripts\Console.py", line 27
    , in <module>
        exec(code, m.__dict__)
      File "C:\DOCUME~1\fabe\LOCALS~1\Temp\tmpcxty5d3f\scripts\dataserv-client.py",
    line 10, in <module>
      File "c:\python\32-bit\3.4\lib\importlib\_bootstrap.py", line 2284, in _handle
    _fromlist
      File "c:\python\32-bit\3.4\lib\importlib\_bootstrap.py", line 321, in _call_wi
    th_frames_removed
      File "c:\python\32-bit\3.4\lib\importlib\_bootstrap.py", line 2237, in _find_a
    nd_load
      File "c:\python\32-bit\3.4\lib\importlib\_bootstrap.py", line 2226, in _find_a
    nd_load_unlocked
      File "c:\python\32-bit\3.4\lib\importlib\_bootstrap.py", line 1191, in _load_u
    nlocked
      File "c:\python\32-bit\3.4\lib\importlib\_bootstrap.py", line 1161, in _load_b
    ackward_compatible
      File "E:\dataserv-client\dataserv_client\cli.py", line 4, in <module>
        from dataserv_client import api
      File "c:\python\32-bit\3.4\lib\importlib\_bootstrap.py", line 2284, in _handle
    _fromlist
      File "c:\python\32-bit\3.4\lib\importlib\_bootstrap.py", line 321, in _call_wi
    th_frames_removed
      File "c:\python\32-bit\3.4\lib\importlib\_bootstrap.py", line 2237, in _find_a
    nd_load
      File "c:\python\32-bit\3.4\lib\importlib\_bootstrap.py", line 2226, in _find_a
    nd_load_unlocked
      File "c:\python\32-bit\3.4\lib\importlib\_bootstrap.py", line 1191, in _load_u
    nlocked
      File "c:\python\32-bit\3.4\lib\importlib\_bootstrap.py", line 1161, in _load_b
    ackward_compatible
      File "E:\dataserv-client\dataserv_client\api.py", line 16, in <module>
        from dataserv_client import builder
      File "c:\python\32-bit\3.4\lib\importlib\_bootstrap.py", line 2284, in _handle
    _fromlist
      File "c:\python\32-bit\3.4\lib\importlib\_bootstrap.py", line 321, in _call_wi
    th_frames_removed
      File "c:\python\32-bit\3.4\lib\importlib\_bootstrap.py", line 2237, in _find_a
    nd_load
      File "c:\python\32-bit\3.4\lib\importlib\_bootstrap.py", line 2226, in _find_a
    nd_load_unlocked
      File "c:\python\32-bit\3.4\lib\importlib\_bootstrap.py", line 1191, in _load_u
    nlocked
      File "c:\python\32-bit\3.4\lib\importlib\_bootstrap.py", line 1161, in _load_b
    ackward_compatible
      File "E:\dataserv-client\dataserv_client\builder.py", line 3, in <module>
        import RandomIO
      File "c:\python\32-bit\3.4\lib\importlib\_bootstrap.py", line 2237, in _find_a
    nd_load
      File "c:\python\32-bit\3.4\lib\importlib\_bootstrap.py", line 2224, in _find_a
    nd_load_unlocked
    ImportError: No module named 'RandomIO'

    E:\dataserv-client\dist\dataserv-client-1.2.1.win32>pip freeze
    You are using pip version 6.0.8, however version 7.1.0 is available.
    You should consider upgrading via the 'pip install --upgrade pip' command.
    cannot determine version of editable source in e:\dataserv-client (git command n
    ot found in path)
    cx-Freeze==4.3.4
    -e dataserv-client==1.2.1
    esky==0.9.9.dev0
    future==0.15.0
    pycrypto==2.6.1
    randomio==0.2.1

    E:\dataserv-client\dist\dataserv-client-1.2.1.win32>python --version
    Python 3.4.3

**Comment by timeyyy** [comment]
I repeat, i had to also install randomIO and hashlib or something like that!, please read what i write mate. If you read the error message it tells you clearly

 ImportError: No module named 'RandomIO',

 you have randomio not RandomIO isntalled..

 pip install RandomIO,

 compile again, you will get another message, so install that dependency as well,

 repeat process.

**Comment by F483** [comment]
I do indeed have RandomIO installed as you can see in the output of `pip freeze` above.

If it is working for you I would like to know the exact setup you have, so I can replicate it. Could you please post your output for python --version, pip --freeze and what version of windows you are using.

**Comment by timeyyy** [comment]
ImportError: No module named 'RandomIO'
it's case sensitive so..
RandomIO==0.2.1

**Comment by timeyyy** [comment]
i deleted the repo after it ran but i'm guessing even though pypi projects are case insensitive it's somehow playing a roll here, pip freeze gives me 

RandomIO==0.2.1

try installing the package as RandomIO

im using
python3.4.3
windows 7
pip 7

**Comment by timeyyy** [comment]
how are you going mate?

**Comment by F483** [comment]
I just setup a Windows 7 VM and tried building it there. I was successful, but only after I uninstallded randomio (was automatically installed dependency via `python setup.py develop`) and installed it manually with `pip install RandomIO`. Why the dependency is listed in a different case or why it should even matter I do not know. I also used cx-Freeze so I'm not sure about the status of py2exe, maybe I will try it and report back.

I suggest the following:

 - First update the documentation, so it mentions all installed packages must be listed in the correct case (reinstall manually to fix it) and to use the latest esky from github instead of pypi when trying to build on windows.
 - Open another issue for esky and get the have the case bug fixed so that case of the package name does not matter.

**Comment by timeyyy** [comment]
i'm glad you got it working, Feel free to help out!

**Comment by F483** [comment]
I'm not sure if I have the time to help another open source project, as I am working on quite a few already. We will however be using esky for https://github.com/Storj/dataserv-client and provide constant feedback.

I will close this Issue as soon as another regarding the case bug is opened.

**Comment by rfk** [comment]
Thanks for reaching out.  Unfortunately this project is no longer actively maintained, so I'm going to move it into archive mode:

  https://rfk.id.au/blog/entry/archiving-open-source-projects/
